youtube_dl/extractor/sohu.py

```diff
@@ -77,6 +104,11 @@ def _fetch_data(vid_id, mytv=False):
                     % (format_id, i + 1, part_count))
 
                 part_info = part_str.split('|')
+
+                # Sanitize URL to prevent download failure
+                if part_info[0][-1] == '/' and su[i][0] == '/':
+                    su[i] = su[i][1:]
+
                 video_url = '%s%s?key=%s' % (part_info[0], su[i], part_info[3])
```

Comment:
I think the proper way to do that is `'%s?key=%s'  % (compat_urlparse.urljoin(part_info[0], su[i]), part_info[3])`.